src/mesh/unstructured_mesh.C

```diff
@@ -116,9 +119,15 @@ void UnstructuredMesh::copy_nodes_and_elements(const UnstructuredMesh & other_me
                                   oldn->get_extra_integer(i));
 
 #ifdef LIBMESH_ENABLE_UNIQUE_ID
-        newn->set_unique_id(oldn->unique_id() + unique_id_offset);
-#endif
+        const unique_id_type new_unique_id = oldn->unique_id() + unique_id_offset;
+        newn->set_unique_id(new_unique_id);
+        potentially_next_unique_id = std::max(new_unique_id+1, potentially_next_unique_id);
       }
+
+    this->set_next_unique_id(potentially_next_unique_id);
```

Comment:
This isn't safe on a distributed `other_mesh`, is it?  If we can't see the whole node range on every processor, we might be setting a value here that's inconsistent between processors and too low on some of them.

I think we're going to need to just call `other_mesh.parallel_max_unique_id()` and add `unique_id_offset` to that.

Boy.  We really ought to add a `this->comm().verify(id)` inside `DistributedMesh::set_next_unique_id` too.  This seems like a really easy mistake to make and a really hard mistake to catch.

And going one more level up the root cause tree ... we're just woefully neglecting `MeshBase::_is_prepared`, aren't we?  I suspect that `copy_nodes_and_elements` works the way it does because we were expecting code using it to `prepare_for_use` afterwards, but with nothing doing `set_isnt_prepared()` here how are we to know that, and with next to nothing asserting `is_prepared()` elsewhere how are we to verify that?

Reply:
...

I'm apparently the confused one.

We do have `prepare_for_use()` at the end of `copy_nodes_and_elements()`, don't we?  The problem is that, while we're always calling `update_parallel_id_counts()` via `prepare_for_use()`, in the guts of `parallel_max_unique_id()` there we still expect to see local values of `_next_unique_id` there which are locally valid, even if they're not globally valid; we don't actually inspect DofObject `unique_id()` values to see if any of them have gotten ahead of us.  And we shouldn't change that; I'd certainly rather not add an additional element loop and node loop inside what was previously an O(1) method.

So, yeah, the solution is still going to be to fix that manually with `other_mesh.parallel_max_unique_id()` plus `unique_id_offset`; my clever ideas about `prepare_for_use()` weren't clever at all.